src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -365,35 +386,44 @@ private static string GetStatusBarText(string packageName, string? installedVers
             => installedVersion == null ? packageName : $"{packageName} - {installedVersion}";
 
         private bool TryUninstallPackage(
-            string packageName, Guid projectGuid, EnvDTE.DTE dte, EnvDTE.Project dteProject)
+            string packageName, Guid projectGuid, EnvDTE.DTE dte, EnvDTE.Project dteProject,
+            IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
             this.AssertIsForeground();
             Contract.ThrowIfFalse(IsEnabled);
 
-            // TODO: consider putting this under a TWD so it would actually be cancellable.
-            var cancellationToken = CancellationToken.None;
+            var description = string.Format(ServicesVSResources.Uninstalling_0, packageName);
+            progressTracker.Description = description;
+            dte.StatusBar.Text = description;
 
             try
             {
-                return this.ThreadingContext.JoinableTaskFactory.Run(() =>
-                {
-                    return this.PerformNuGetProjectServiceWorkAsync(async (nugetService, cancellationToken) =>
+                return this.ThreadingContext.JoinableTaskFactory.Run(
+                    () => this.PerformNuGetProjectServiceWorkAsync(async (nugetService, cancellationToken) =>
                     {
-                        await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
Same feedback as above: aren't you still blocking the main thread by virtue of being inside a JTF.Run that we know was called from the main thread?

Reply:
yeah, i will fix this all up :)